File: app/src/main/java/br/com/gabryel/financeapp/investment/SimpleWallet.java
package br.com.gabryel.financeapp.investment;

import java.util.ArrayList;
import java.util.List;

/**
 * Simple implementation of Wallet class.
 * <p/>
 * Created by gabryel on 05/04/16.
 */
public class SimpleWallet implements Wallet {

	private String name;

	private List<Moneyable> investments = new ArrayList<>();

	public SimpleWallet(String name) {
		this.name = name;
	}

	@Override
	public void add(Moneyable investment) {
		investments.add(investment);
	}

	@Override
	public String getName() {
		return name;
	}
}

File: app/src/main/java/br/com/gabryel/financeapp/investment/SimpleWallet.kt
package br.com.gabryel.financeapp.investment

/**
 * Simple implementation of Wallet class.
 *
 *
 * Created by gabryel on 05/04/16.
 */
class SimpleWallet(override val name: String) : Wallet {

    private val investments = mutableListOf<Moneyable>()

    override fun add(investment: Moneyable) {
        investments.add(investment)
    }
}